core/unit_test/TestGraph.hpp

```diff
@@ -103,28 +103,37 @@ struct TEST_CATEGORY_FIXTURE(graph) : public ::testing::Test {
   }
 };
 
+// Check if a rank-0 view contains a given value.
+template <typename Exec, typename ViewType>
+::testing::AssertionResult contains(const Exec& exec, const ViewType& view, const typename ViewType::value_type& expected) {
+  static_assert(ViewType::rank == 0);
+  typename ViewType::non_const_value_type value;
+  Kokkos::deep_copy(exec, value, view);
+  exec.fence();
+  if(value != expected)
+    return ::testing::AssertionFailure() << expected << " is not in " << view.label();
+  else
+    return ::testing::AssertionSuccess();
```

Comment:
What does the final output on failure look like?

Reply:
```bash
[ RUN      ] cuda_graph.submit_once_rvalue
/workspaces/kokkos/core/unit_test/TestGraph.hpp:135: Failure
Value of: contains(ex, count, 5)
  Actual: false (5 is not in count, got 1)
Expected: true
[  FAILED  ] cuda_graph.submit_once_rvalue (0 ms)
```

Reply:
is it critical that the execution space is specified?
contains -> rank_0_view_contains?

Reply:
I guess it is critical, because if your test is asynchronicity-critical and `contains` makes a global fence, your test might pass thought something is semantically wrong, right ?

Reply:
For the naming, I think we could keep `contains` and better constraint it so that it's clear from the signature it's for rank-0 views ? (We could wait for `C++20` to constraint the `ViewType` directory)

Reply:
@dalg24 OK to close ?